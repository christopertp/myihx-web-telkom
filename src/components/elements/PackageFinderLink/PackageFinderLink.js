import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './styles.scoped.css';

export default function PackageFinderLink(props) {
  const { className } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  return (
    <section className={customClass}>
      <Link to="#">
        <img alt="grfx find" src="/assets/grfx_find.png" />
        <span>Perlu bantuan memilih paket internet? Coba package finder kami</span>
        <img alt="chevron white" src="/assets/ic_chevron_white.svg" />
      </Link>
    </section>
  );
}

PackageFinderLink.defaultProps = {
  className: '',
};

PackageFinderLink.propTypes = {
  className: PropTypes.string,
};
