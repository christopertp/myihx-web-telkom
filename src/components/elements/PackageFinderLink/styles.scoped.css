.root {
  margin: 0 var(--neg-side-margin);
  padding: 0 0 3rem;
}

.root > a {
  align-items: center;
  background: url(/assets/bnr_blue_2nd.png) center center no-repeat;
  background-size: cover;
  border-radius: 0.5rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  color: #ffffff;
  display: flex;
  font-size: 1.125rem;
  font-weight: 800;
  line-height: 1.2;
  margin: 0 auto;
  padding: 1.25rem 1rem 1rem;
  text-align: left;
  width: 45.875rem;
}

.root > a > span {
  width: 21rem;
}

.root > a > img:first-of-type {
  height: 4.6875rem;
  margin-right: 1rem;
}

.root > a > img:last-of-type {
  height: 1.5rem;
  margin-left: auto;
}

@media only screen and (max-width: 48em) {
  .root {
    background-color: #f2f2f2;
    margin: 0 -1rem;
    padding: 0 1rem 1rem;
  }

  .root > a {
    background-size: auto;
    font-size: 0.7rem;
    padding: 0.7rem 1rem;
    width: 100%;
  }

  .root > a > span {
    font-size: 1rem;
    width: auto;
  }

  .root > a > img:first-of-type {
    height: 4rem;
  }
}
