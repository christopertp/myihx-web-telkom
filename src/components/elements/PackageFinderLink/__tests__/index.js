import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PackageFinderLink from '../PackageFinderLink';

describe('src/components/elements/PackageFinderLink', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PackageFinderLink />);
    expect(tree).toMatchSnapshot();
  });
});
