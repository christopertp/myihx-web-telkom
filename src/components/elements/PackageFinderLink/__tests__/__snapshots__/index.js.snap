// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/PackageFinderLink render 1`] = `
<section
  className=""
>
  <Link
    to="#"
  >
    <img
      alt="grfx find"
      src="/assets/grfx_find.png"
    />
    <span>
      Perlu bantuan memilih paket internet? Coba package finder kami
    </span>
    <img
      alt="chevron white"
      src="/assets/ic_chevron_white.svg"
    />
  </Link>
</section>
`;
